File: python/src/loadstrike_sample_reference/features/transport_http/http_load_shared_client_support.py
from __future__ import annotations

from requests import Session
from requests.adapters import HTTPAdapter

from loadstrike_sdk import LoadStrikeResponse

_shared_session = Session()
_shared_adapter = HTTPAdapter(pool_connections=128, pool_maxsize=1000, max_retries=0)
_shared_session.mount("https://", _shared_adapter)
_shared_session.mount("http://", _shared_adapter)
_shared_session.headers.update({"Accept-Encoding": "gzip, deflate"})


def post_order(order_id: str, tenant_id: str):
    response = _shared_session.post(
        "https://api.example.com/orders",
        json={
            "orderId": order_id,
            "tenantId": tenant_id,
            "amount": 49.95,
        },
        timeout=15,
    )

    if response.ok:
        return LoadStrikeResponse.ok(str(response.status_code))
    return LoadStrikeResponse.fail(str(response.status_code))


def close_shared_session() -> None:
    _shared_session.close()

File: js/src/features/transport_http/http-load-shared-client-support.mjs
import { Agent, fetch } from "undici";
import { LoadStrikeResponse } from "@loadstrike/loadstrike-sdk";

export const sharedHttpClient = new Agent({
  connections: 1000,
  keepAliveTimeout: 120_000,
  keepAliveMaxTimeout: 300_000,
  connectTimeout: 15_000,
  pipelining: 1,
});

export async function postOrder(orderId, tenantId) {
  const response = await fetch("https://api.example.com/orders", {
    method: "POST",
    headers: {
      "content-type": "application/json",
      "accept-encoding": "gzip, deflate",
    },
    body: JSON.stringify({
      orderId,
      tenantId,
      amount: 49.95,
    }),
    dispatcher: sharedHttpClient,
  });

  await response.arrayBuffer();

  return response.ok
    ? LoadStrikeResponse.ok(String(response.status))
    : LoadStrikeResponse.fail(String(response.status));
}

export async function closeSharedHttpClient() {
  await sharedHttpClient.close();
}
